### ISSUE TITLE
Restore the ability to do raw API requests using the request() method

### ISSUE THREAD

**Issue by Berdir** [thinkshout/mailchimp-api-php#112]
In 2.x, you could call the request method directly, this was quite useful for things that wasn't supported by API methods, example:

```php
$mailchimp_api = mailchimp_get_api_object('MailchimpLists');
$mailchimp_api->request('DELETE', '/lists/{list_id}/interest-categories/{interest_category_id}/interests/{interest_id}', $tokens);
```

Because that method was moved down into the encapsulated API class, that's no longer possible, and you can't access that API object, the factory only creates it inline too.

Suggestions:
- Re-add a public request() method that calls the internal on. Advantage would be that it is not a BC break for existing code.
- Add a method to get the API object from these objects

Alternatively, the Drupal module could extract the part that creates the API object in \Drupal\mailchimp\ClientFactory::createInstance() to a separate method, so you can use getApi() or something on it.

**Comment by aprice42** [comment]
Thanks for raising this issue @Berdir, I think this can be a really simple fix, though it does create a bit of redundancy. But that feels ok for now so we can continue to support both api_key and OAuth authentication for the time being.

I created a PR with a fix you proposed if you want to take a look and provide any feedback. #113

**Comment by aprice42** [comment]
This is resolved now by #113

**Comment by Berdir** [comment]
There is a bug with this that I didn't spot in my cursory review, see https://github.com/thinkshout/mailchimp-api-php/issues/114
